File: BackEnd.v2/demo/src/main/java/com/example/demo/models/SolarPanel.java
package com.example.demo.models;


import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonIdentityReference;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import jakarta.persistence.*;
import lombok.Data;

import java.util.UUID;

@Data
@Entity
@Table(name = "solarPanel")
public class SolarPanel {

    @Id
    @GeneratedValue
    private UUID id;
    private String model;
    private Float nominal_power;
    private Float maximum_voltage;
    private Float maximum_current;
    private Float height;
    private Float width;
    private Float weight;
    private String type_cell;
    private Float price;
    private Integer quantity;
    @Lob
    @Column(name = "image", length = 500000000)
    private String image;
    @JsonIdentityInfo(generator= ObjectIdGenerators.PropertyGenerator.class, property="id")
    @JsonIdentityReference(alwaysAsId=true)
    @ManyToOne
    private Company company;
}

File: FrontEnd.v2/src/app/models/solar-panel.ts
/* tslint:disable */
/* eslint-disable */
import {Company} from "./Company";

export interface SolarPanel {
  companyDto: Company;
  height: number;
  id?: any;
  image: string;
  maximum_current: number;
  maximum_voltage: number;
  model: string;
  nominal_power: number;
  price: number;
  quantity: number;
  type_cell: string;
  weight: number;
  width: number;
}
